source/extensions/stat_sinks/common/statsd/statsd.cc

```diff
@@ -61,19 +62,57 @@ void UdpStatsdSink::flush(Stats::MetricSnapshot& snapshot) {
 
   for (const auto& counter : snapshot.counters()) {
     if (counter.counter_.get().used()) {
-      const std::string counter_str =
-          absl::StrCat(prefix_, ".", getName(counter.counter_.get()), ":", counter.delta_, "|c",
-                       buildTagStr(counter.counter_.get().tags()));
-      writeBuffer(buffer, writer, counter_str);
+      switch (tag_format_.tag_position) {
+      case Statsd::TagPosition::TagAfterValue: {
+        const std::string counter_str = absl::StrCat(
+            // metric name
+            prefix_, ".", getName(counter.counter_.get()),
+            // value and type
+            ":", counter.delta_, "|c",
+            // tags
+            buildTagStr(counter.counter_.get().tags()));
+        writeBuffer(buffer, writer, counter_str);
+      } break;
+
+      case Statsd::TagPosition::TagAfterName: {
+        const std::string counter_str = absl::StrCat(
+            // metric name
+            prefix_, ".", getName(counter.counter_.get()),
+            // tags
+            buildTagStr(counter.counter_.get().tags()),
+            // value and type
+            ":", counter.delta_, "|c");
+        writeBuffer(buffer, writer, counter_str);
+      } break;
+      }
```

Comment:
This logic is duplicated in 3 places, please collapse this into a utility somehow.